File: voiceengine/src/main/java/com/backdoor/engine/lang/WorkerInterface.java
package com.backdoor.engine.lang;

import com.backdoor.engine.misc.Action;
import com.backdoor.engine.misc.Ampm;
import com.backdoor.engine.misc.Long;

import java.util.List;

public interface WorkerInterface {
    boolean hasCalendar(String input);

    String clearCalendar(String input);

    List<Integer> getWeekDays(String input);

    String clearWeekDays(String input);

    long getDaysRepeat(String input);

    String clearDaysRepeat(String input);

    boolean hasRepeat(String input);

    String clearRepeat(String input);

    boolean hasTomorrow(String input);

    String clearTomorrow(String input);

    String getMessage(String input);

    String clearMessage(String input);

    Action getMessageType(String input);

    String clearMessageType(String input);

    Ampm getAmpm(String input);

    String clearAmpm(String input);

    long getTime(String input, Ampm ampm, String[] times);

    String clearTime(String input);

    String getDate(String input, Long res);

    boolean hasCall(String input);

    String clearCall(String input);

    boolean isTimer(String input);

    String cleanTimer(String input);

    boolean hasSender(String input);

    String clearSender(String input);

    boolean hasNote(String input);

    String clearNote(String input);

    boolean hasAction(String input);

    Action getAction(String input);

    boolean hasEvent(String input);

    Action getEvent(String input);

    String getMultiplier(String input, Long res);

    String replaceNumbers(String input);

    boolean hasDisableReminders(String input);

    boolean hasEmptyTrash(String input);

    boolean hasGroup(String input);

    String clearGroup(String input);

    boolean hasToday(String input);

    String clearToday(String input);

    boolean hasAfterTomorrow(String input);

    String clearAfterTomorrow(String input);

    boolean hasAnswer(String input);

    Action getAnswer(String input);

    boolean hasShowAction(String input);

    Action getShowAction(String input);

    boolean hasNextModifier(String input);

    boolean hasEveryDay(String input);
}

File: voice-engine-ktx/src/main/java/com/backdoor/engine/lang/WorkerInterface.kt
package com.backdoor.engine.lang

import com.backdoor.engine.misc.Action
import com.backdoor.engine.misc.Ampm
import com.backdoor.engine.misc.LongInternal
import org.threeten.bp.LocalDate
import org.threeten.bp.LocalTime

internal interface WorkerInterface {
  fun splitWords(input: String?): String?
  fun replaceNumbers(input: String?): String?
  fun hasCalendar(input: String): Boolean
  fun clearCalendar(input: String): String?
  fun getWeekDays(input: String): List<Int>
  fun clearWeekDays(input: String): String
  fun getDaysRepeat(input: String): Long
  fun clearDaysRepeat(input: String): String?
  fun clearRepeat(input: String): String?
  fun hasTomorrow(input: String): Boolean
  fun clearTomorrow(input: String): String?
  fun getMessage(input: String): String
  fun clearMessage(input: String): String?
  fun clearMessageType(input: String): String?
  fun clearAmpm(input: String): String?
  fun getTime(input: String, ampm: Ampm?, times: List<String>): LocalTime?
  fun clearTime(input: String?): String
  fun getDateAndClear(input: String, result: (LocalDate?) -> Unit): String?
  fun clearCall(input: String): String?
  fun cleanTimer(input: String): String?
  fun hasSender(input: String): Boolean
  fun clearSender(input: String): String?
  fun findSenderAndClear(input: String, action: Action, result: (String) -> Unit): String
  fun hasNote(input: String): Boolean
  fun clearNote(input: String): String
  fun hasAction(input: String): Boolean
  fun getAction(input: String): Action?
  fun hasEvent(input: String): Boolean
  fun getEvent(input: String): Action?
  fun getMultiplier(input: String, res: LongInternal): String
  fun hasDisableReminders(input: String): Boolean
  fun hasEmptyTrash(input: String): Boolean
  fun hasGroup(input: String): Boolean
  fun clearGroup(input: String): String
  fun hasToday(input: String): Boolean
  fun clearToday(input: String): String
  fun hasAfterTomorrow(input: String): Boolean
  fun clearAfterTomorrow(input: String): String
  fun hasAnswer(input: String): Boolean
  fun getAnswer(input: String): Action?
  fun hasShowAction(input: String): Boolean
  fun getShowAction(input: String): Action?
  fun clearShowAction(input: String): String
  fun hasNextModifier(input: String): Boolean
  fun hasCall(input: String): Boolean
  fun hasRepeat(input: String): Boolean
  fun hasEveryDay(input: String): Boolean
  fun getMessageType(input: String): Action?
  fun getAmpm(input: String): Ampm?
  fun hasTimer(input: String): Boolean
}
